### ISSUE TITLE
training loss dosen't drop

### ISSUE THREAD

**Issue by CLeaf1023** [om-ai-lab/VLM-R1#116]
I also have same problem. I use 3A800 to train. 
this is my script
`cd src/open-r1-multimodal

export DEBUG_MODE="true"
export CUDA_VISIBLE_DEVICES=4,5,7

RUN_NAME="Qwen2.5-VL-3B-GRPO-REC-0303"
export LOG_PATH="./debug_log_$RUN_NAME.txt"

torchrun --nproc_per_node="2" \
    --nnodes="1" \
    --node_rank="0" \
    --master_addr="127.0.0.1" \
    --master_port="12346" \
    src/open_r1/grpo_rec.py \
    --deepspeed local_scripts/zero3.json \
    --output_dir output/$RUN_NAME \
    --model_name_or_path Qwen/Qwen2.5-VL-3B-Instruct \
    --dataset_name data_config/rec.yaml \
    --image_root /data/cuiyajie/VLM-R1-main/dataset \
    --max_prompt_length 1024 \
    --num_generations 8 \
    --per_device_train_batch_size 2 \
    --gradient_accumulation_steps 2 \
    --logging_steps 1 \
    --bf16 \
    --torch_dtype bfloat16 \
    --data_seed 42 \
    --report_to tensorboard \
    --gradient_checkpointing true \
    --attn_implementation flash_attention_2 \
    --num_train_epochs 2 \
    --run_name $RUN_NAME \
    --save_steps 100 \
    --save_only_model true
`
the reward curve I got:

![Image](https://github.com/user-attachments/assets/bfdcad65-8cfe-4e7b-baa5-ab1003fe57f2)

the training loss curve:

![Image](https://github.com/user-attachments/assets/385373cf-6d18-41e8-81c2-a175a54bb128)

is there anything I missed or got messed up? I think the curve is abnormal

**Comment by gzyao** [comment]
May I ask if the dataset you are training on is customized or provided by the official coco?

**Comment by cyj95** [comment]
> May I ask if the dataset you are training on is customized or provided by the official coco?

official coco

**Comment by liuw20** [comment]
I have same question and  my curve is like this:

![Image](https://github.com/user-attachments/assets/78cb6aa1-81b9-47db-a7c4-04aa1e662116)

And I found my loss has no degration.

**Comment by z666-123** [comment]
I am training with 6000 samples extracted from the dataset refcoco_train.json provided by this project, using the script run_grpo_rec_lora.sh. Initially, I trained using the original script and noticed significant fluctuations. Therefore, I lowered the learning rate and used 10% of the steps as a warm-up period. Could you please tell me if the visualization metrics are normal? If not, where might the problem lie? I have the following concerns: 1. The iou_reward seems to fluctuate a lot. 2. The format reward tends to 1 in the later stages, which looks normal. 3. There was an abnormal peak in training loss. 4. The KL divergence also had a similar abnormal peak at one point.


```
cd src/open-r1-multimodal

export DEBUG_MODE="true"
export CUDA_VISIBLE_DEVICES=3,5

# RUN_NAME="Qwen2.5-VL-7B-GRPO-REC-lora"
RUN_NAME="Qwen2.5-VL-3B-GRPO-REC-lora-COCO_01_3_16_smooth"
export LOG_PATH="./debug_log_$RUN_NAME.txt"

torchrun --nproc_per_node="2" \
    --nnodes="1" \
    --node_rank="0" \
    --master_addr="127.0.0.1" \
    --master_port="12346" \
    src/open_r1/grpo_rec.py \
    --deepspeed local_scripts/zero2.json \
    --output_dir output/$RUN_NAME \
    --model_name_or_path models/Qwen2.5-VL-3B-Instruct \
    --dataset_name data_config/rec.yaml \
    --image_root data/ \
    --max_prompt_length 1024 \
    --num_generations 8 \
    --per_device_train_batch_size 4 \
    --gradient_accumulation_steps 4 \
    --logging_steps 1 \
    --bf16 \
    --torch_dtype bfloat16 \
    --data_seed 42 \
    --report_to wandb \
    --gradient_checkpointing true \
    --attn_implementation flash_attention_2 \
    --num_train_epochs 1 \
    --run_name $RUN_NAME \
    --save_steps 100 \
    --save_only_model true \
    --learning_rate 5e-6 \
    --use_peft true \
    --lora_r 64 \
    --lora_alpha 128 \
    --lora_dropout 0.1 \
    --lora_task_type CAUSAL_LM \
    --freeze_vision_modules true \
    --resume_from_checkpoint None \
    --warmup_ratio 0.1 \
    --lr_scheduler_type cosine
    # 使用10%的步数作为预热期
    # 使用余弦学习率调度
    # output/Qwen2.5-VL-3B-GRPO-REC-lora-COCO_01/checkpoint-100

```

![Image](https://github.com/user-attachments/assets/942f5d31-510a-4304-a5ad-22230ae391b8)

![Image](https://github.com/user-attachments/assets/ed645562-447c-44b6-89a9-77d553515884)

**Comment by Shengqi77** [comment]
Hi！
How do you use TensorBoard to display training information? Besides setting report_to='tensorboard', are there other configurations needed in the code? Could you share an example of your TensorBoard training visualization code?

**Comment by LijunRio** [comment]
I also get the same problem, the training loss dosen't drop

![Image](https://github.com/user-attachments/assets/904c657a-3e3f-43be-bc0c-48da2f2a864f)
